## 1. BUSINESS UNDERSTANDING

In [ ]:
# Importing the initial libraries needed for the project
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [ ]:
# Reading the product data
products = pd.read_csv("products.csv")

products.head(2)

Uniq_id           SKU  \
0  b6c0b6bea69c722939585baeac73c13d  pp5006380337   
1  93e5272c51d8cce02597e3ce67b7ad0a  pp5006380337   

                                          Name  \
0  Alfred Dunner® Essential Pull On Capri Pant   
1  Alfred Dunner® Essential Pull On Capri Pant   

                                         Description  Price  Av_Score  
0  Youll return to our Alfred Dunner pull-on capr...  41.09     2.625  
1  Youll return to our Alfred Dunner pull-on capr...  41.09     3.000

I want to learn more about the data which can help my analysis

In [ ]:
# Show the number of rows and columns in the 'products' DataFrame
products.shape

(7982, 6)

In [ ]:
# Show summary information about the DataFrame, including column data types and missin
products.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7982 entries, 0 to 7981
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Uniq_id      7982 non-null   object 
 1   SKU          7915 non-null   object 
 2   Name         7982 non-null   object 
 3   Description  7439 non-null   object 
 4   Price        5816 non-null   float64
 5   Av_Score     7982 non-null   float64
dtypes: float64(2), object(4)
memory usage: 374.3+ KB


In [ ]:
# Display summary statistics (like mean, min, max, and quartiles) for the numerical columns in the DataFrame
products.describe()


Price     Av_Score
count   5816.000000  7982.000000
mean     144.776618     2.988683
std      499.223719     0.911673
min      -65.270000     1.000000
25%       40.700000     2.500000
50%       58.010000     3.000000
75%       87.020000     3.500000
max    17122.170000     5.000000

In [ ]:
# Count the number of missing (null) values in each column of the DataFrame
products.isnull().sum()

Uniq_id           0
SKU              67
Name              0
Description     543
Price          2166
Av_Score          0
dtype: int64

I want to explore the other datasets before I know my next step

In [ ]:
# Reading the reviews data
reviews = pd.read_csv("reviews.csv")
reviews.head(2)

Uniq_id  Username  Score  \
0  b6c0b6bea69c722939585baeac73c13d  fsdv4141      2   
1  b6c0b6bea69c722939585baeac73c13d  krpz1113      1   

                                              Review  
0  You never have to worry about the fit...Alfred...  
1  Good quality fabric. Perfect fit. Washed very ...

In [ ]:
# Show the number of rows and columns in the 'reviews' DataFrame
reviews.shape


(39063, 4)

In [ ]:
# Show summary information about the DataFrame, including column data types and missin
reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39063 entries, 0 to 39062
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Uniq_id   39063 non-null  object
 1   Username  39063 non-null  object
 2   Score     39063 non-null  int64 
 3   Review    39063 non-null  object
dtypes: int64(1), object(3)
memory usage: 1.2+ MB


In [ ]:
# Display summary statistics (like mean, min, max, and quartiles) for the numerical columns in the DataFrame
reviews.describe()

Score
count  39063.000000
mean       1.487648
std        1.400332
min        0.000000
25%        0.000000
50%        1.000000
75%        2.000000
max        5.000000

In [ ]:
# Count the number of missing (null) values in each column of the DataFrame
reviews.isnull().sum()

Uniq_id     0
Username    0
Score       0
Review      0
dtype: int64

In [ ]:
# Reading the users data
users = pd.read_csv("users.csv")
users.head(2)

Username         DOB          State
0  bkpn1412  31.07.1983         Oregon
1  gqjs4414  27.07.1998  Massachusetts

In [ ]:
# Show the number of rows and columns in the 'products' DataFrame
users.shape

(5000, 3)

In [ ]:
# Show summary information about the DataFrame, including column data types and missin
users.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Username  5000 non-null   object
 1   DOB       5000 non-null   object
 2   State     5000 non-null   object
dtypes: object(3)
memory usage: 117.3+ KB


In [ ]:
# Display summary statistics (like mean, min, max, and quartiles) for the numerical columns in the DataFrame
users.describe()

Username         DOB          State
count       5000        5000           5000
unique      4999          52             57
top     dqft3311  07.08.1953  Massachusetts
freq           2         112            107

In [ ]:
# Count the number of missing (null) values in each column of the DataFrame
users.isnull().sum()

Username    0
DOB         0
State       0
dtype: int64

The products DataFrame contains 7982 rows and 6 columns. It has significant missing values in SKU (67), Description (543), and Price (2166). Notably, the Price column has an unusual minimum value of -65.27. The average score ranges from 1.0 to 5.0, with a mean of 2.99.

The reviews DataFrame contains 39063 rows and 4 columns, with no missing values. The Score column ranges from 0 to 5, with a mean of 1.49, which is considerably lower than the Av_Score observed in the products DataFrame.

The users DataFrame contains 5000 rows and 3 columns, with no missing values. The DOB column is stored as an object (string) and not a datetime. There is a duplicate username and 57 unique states, with 'Massachusetts' being the most frequent.

I want to extract the age from the date of birth

In [ ]:
users['DOB'] = pd.to_datetime(users['DOB'], format='%d.%m.%Y')
users['age'] = (pd.to_datetime('now').year - users['DOB'].dt.year)
users.head(2)

Username        DOB          State  age
0  bkpn1412 1983-07-31         Oregon   42
1  gqjs4414 1998-07-27  Massachusetts   27

In [ ]:
# handling negative prices in price column
products["Price"] = products["Price"].abs()

In [ ]:
products.isnull().sum()

Uniq_id           0
SKU              67
Name              0
Description     543
Price          2166
Av_Score          0
dtype: int64

The price still has null values. I will replace them with the median of the prices of products that have the same SKU

In [ ]:
products['Price'] = products.groupby('SKU')['Price'].transform(lambda x: x.fillna(x.median()))

In [ ]:
products.isnull()

Uniq_id    SKU   Name  Description  Price  Av_Score
0       False  False  False        False  False     False
1       False  False  False        False  False     False
2       False  False  False        False  False     False
3       False  False  False        False  False     False
4       False  False  False        False  False     False
...       ...    ...    ...          ...    ...       ...
7977    False  False  False        False  False     False
7978    False  False  False        False  False     False
7979    False  False  False        False  False     False
7980    False  False  False         True   True     False
7981    False  False  False         True   True     False

[7982 rows x 6 columns]

In [ ]:
products.isnull().sum()

Uniq_id           0
SKU              67
Name              0
Description     543
Price          1955
Av_Score          0
dtype: int64

In [ ]:
products.isnull().sum()

Uniq_id           0
SKU              67
Name              0
Description     543
Price          1955
Av_Score          0
dtype: int64

In [ ]:
products['Price'] = products['Price'].fillna(products['Price'].median())

In [ ]:
products.dropna(subset=['SKU'], inplace=True)

Now that the Data is prepared, I want to start making the visualisations to gather insights. the first I would do is to look for the top 15 products bought in JCPenney